legend-engine-language-pure-store-relational/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DatasourceSpecificationBuilder.java

```diff
@@ -76,6 +76,15 @@ else if(datasourceSpecification instanceof BigQueryDatasourceSpecification)
             _bigquery._defaultDataset(bigQueryDatasourceSpecification.defaultDataset);
             return _bigquery;
         }
+        else if(datasourceSpecification instanceof SpannerDatasourceSpecification)
```

Comment:
Can we add a test for compilation too?

Reply:
Sure. Like TestRelationalConnectionCompilationRoundtrip.testConnectionPropertiesPropagatedToCompiledGraph ?

Reply:
Yes something like that
So I realize adding asserts on pure model objects is not convenient but as of now we don't have a roundtrip test for compiler (atleast I am not aware of one) so we can add a similar test or trying adding logic for roundtrip tests for compiler